cdc/sink/producer/kafka/metrics.go

```diff
@@ -262,22 +280,42 @@ func (sm *saramaMetricsMonitor) collectBrokerMetrics() error {
 			responseSizeGauge.WithLabelValues(sm.changefeedID, brokerID).Set(histogram.Snapshot().Mean())
 		}
 	}
-	return nil
 }
 
-func newSaramaMetricsMonitor(registry metrics.Registry, changefeedID string, admin kafka.ClusterAdminClient) *saramaMetricsMonitor {
-	return &saramaMetricsMonitor{
+// flushMetricsInterval specifies the interval of refresh sarama metrics.
+const flushMetricsInterval = 5 * time.Second
+
+func runSaramaMetricsMonitor(ctx context.Context, registry metrics.Registry, changefeedID string,
+	role util.Role, admin kafka.ClusterAdminClient) {
+	monitor := &saramaMetricsMonitor{
 		changefeedID: changefeedID,
+		role:         role,
 		registry:     registry,
 		admin:        admin,
+		brokers:      make(map[int32]struct{}),
 	}
+
+	ticker := time.NewTicker(flushMetricsInterval)
+	go func() {
+		defer func() {
+			ticker.Stop()
+			monitor.cleanup()
+		}()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				monitor.collectMetrics()
```

Comment:
since collectMetrics will hang, print warning log if timeout? for example:  check the duration is gt 5s?